src/openshift/cluster.ts

```diff
@@ -103,20 +107,55 @@ export class Cluster extends OpenShiftItem {
 
     @vsCommand('openshift.explorer.switchContext')
     static async switchContext(): Promise<string> {
-        const k8sConfig = new KubeConfigUtils();
-        const contexts = k8sConfig.contexts.filter((item) => item.name !== k8sConfig.currentContext);
-        const contextName: QuickPickItem[] = contexts.map((ctx) => ({ label: `${ctx.name}`}));
-        if (contextName.length === 0) {
-            const command = await window.showInformationMessage('You have no Kubernetes contexts yet, please login to a cluster.', 'Login', 'Cancel');
-            if (command === 'Login') {
-                return Cluster.login(undefined, true);
+        return new Promise<string>((resolve, reject) => {
+            const k8sConfig = new KubeConfigUtils();
+            const contexts = k8sConfig.contexts.filter((item) => item.name !== k8sConfig.currentContext);
+            const deleteBtn = new quickBtn(new ThemeIcon('trash'), 'Delete');
+            const quickPick = window.createQuickPick();
+            const contextNames: QuickPickItem[] = contexts.map((ctx) => ({ label: `${ctx.name}`, buttons: [deleteBtn] }));
+            quickPick.items = contextNames;
+            if (contextNames.length === 0) {
+                void window.showInformationMessage('You have no Kubernetes contexts yet, please login to a cluster.', 'Login', 'Cancel')
+                    .then((command: string) => {
+                        if (command === 'Login') {
+                            resolve(Cluster.login(undefined, true));
+                        }
+                        resolve(null);
+                    })
+            } else {
+                let selection: readonly QuickPickItem[] | undefined;
+                quickPick.onDidChangeSelection((selects) => {
+                    selection = selects;
+                });
+                quickPick.onDidAccept(async () => {
+                    if (selection && selection.length > 0) {
+                        const choice = Array.from(selection).pop();
+                        await Cluster.odo.execute(Command.setOpenshiftContext(choice.label));
+                        resolve(`Cluster context is changed to: ${choice.label}`);
+                    }
+                });
+                quickPick.onDidTriggerItemButton(async (event: QuickPickItemButtonEvent<QuickPickItem>) => {
+                    if (event.button === deleteBtn) {
+                        await window.showInformationMessage('Are you sure you want to delete this cluster information from kubeconfig?', 'Yes', 'No')
```

Comment:
should assign awaited value

`const answer = await ...`

then you do not need to use `then` callback and code would look much cleaner